admin/services/helpdesk.service.js

```diff
@@ -4,17 +4,21 @@ const roles = require('../lib/consts/service-roles.js')
 const helpdeskService = {}
 
 /**
- * Identify if helpdesk user has not received impersonation
+ * Identify if role is helpdesk
  * @param {object} user
  * @returns {boolean}
  */
-helpdeskService.hasHelpdeskNotReceivedImpersonation = (user) => {
-  if (user.role !== roles.helpdesk) {
-    return false
-  }
-  if (!user.School || !user.schoolId || typeof user.timezone !== 'string') {
-    return true
-  }
+helpdeskService.isHelpdeskRole = (user) => {
+  return user.role === roles.helpdesk
+}
+
+/**
+ * Identify if current user has impersonation properties set
+ * @param {object} user
+ * @returns {boolean}
+ */
+helpdeskService.isImpersonating = (user) => {
+  return user.School && user.schoolId && typeof user.timezone === 'string'
```

Comment:
i would just do a typeof check for number on schoolId.  its more explicit

Reply:
@GuyHarwood fair point. amended